src/fabric.c

```diff
@@ -63,6 +63,52 @@ int ofi_init = 0;
 extern struct ofi_common_locks common_locks;
 
 static struct fi_filter prov_filter;
+static struct fi_filter prov_post_filter;
+
+static int ofi_prov_name_match_filter(const char *prov_name, struct fi_filter *filter)
+{
+	int i, len;
+
+	if (!filter || !filter->names || !filter->names[0])
+		return 1;
+
+	for (i=0; filter->names[i]; i++) {
+		len = strlen(filter->names[i]);
+		if (!strncmp(prov_name, filter->names[i], len) &&
+		    (prov_name[len] == '\0' || prov_name[len] ==';'))
```

Comment:
For the post filter, I was expecting a simple string comparison match.  Why are the extra prov_name checks here?

Reply:
the extra check allows extra layering. for example, if user asks for "verbs;ofi_rxm" then both "verbs;ofi_rxm" and "verbs;ofi_rxm;ofi_mrail" will match, but should match "verbs;ofi_rxm2".

Reply:
I didn't even consider the case where the user could specify "verbs;ofi_rxm,verbs;ofi_rxd".  Does the change handle that?

The extra checks make sense now.  Thanks